server/src/auth/auth0.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const auth_utils = require('./utils');
+const logger = require('../logger');
+
+const https = require('https');
+const Joi = require('joi');
+const querystring = require('querystring');
+const url = require('url');
+
+const options_schema = Joi.object().keys({
+  path: Joi.string().required(),
+  id: Joi.string().required(),
+  secret: Joi.string().required(),
+  host: Joi.string().required()
+}).unknown(false);
+
+function auth0(horizon, raw_options) {
+  const options = Joi.attempt(raw_options, options_schema);
+  const response_type = 'token';
+  const client_id = options.id;
+  const client_secret = options.secret;
+  const provider = options.path;
+  const return_url = horizon._auth._success_redirect.href.slice(0,-1);
+  const host = options.host;
+  
+  const self_url = (self_host, path) =>
+    url.format({ protocol: 'https', host: self_host, pathname: path });
+
+  const make_acquire_url = (state, redirect_uri) =>
+    url.format({ protocol: 'https',
+                 host: host,
+                 pathname: '/authorize',
+                 query: { response_type, client_id, redirect_uri, state } });
+
+  const make_inspect_request = (access_token) =>
+    https.request({ host: host,
+                    path: '/userinfo',
+                    headers: { 'Authorization': `Bearer ${access_token}` } });
+
+  const extract_id = (user_info) => user_info && user_info.identities[0].user_id;
+
+  const make_success_url = (horizon_token) =>
+    url.format(auth_utils.extend_url_query(horizon._auth._success_redirect, { horizon_token }));
+
+  const make_failure_url = (horizon_error) =>
+    url.format(auth_utils.extend_url_query(horizon._auth._failure_redirect, { horizon_error }));
+
+  horizon.add_http_handler(provider, (req, res) => {
+    const request_url = url.parse(req.url, true);
+    const access_token = request_url.query && request_url.query.access_token;
```

Comment:
access_token is not parsing cause the req is not getting the part of the url afther the # auth0 builds tokens with a # instead of a ?